src/uu/sort/src/sort.rs

```diff
@@ -322,25 +322,50 @@ struct Precomputed {
 
 impl GlobalSettings {
     /// Parse a SIZE string into a number of bytes.
-    /// A size string comprises an integer and an optional unit.
+    /// A size string comprises an integer and an optional unit or percentage of total memory.
     /// The unit may be k, K, m, M, g, G, t, T, P, E, Z, Y (powers of 1024), or b which is 1.
     /// Default is K.
     fn parse_byte_count(input: &str) -> Result<usize, ParseSizeError> {
         // GNU sort (8.32)   valid: 1b,        k, K, m, M, g, G, t, T, P, E, Z, Y
         // GNU sort (8.32) invalid:  b, B, 1B,                         p, e, z, y
+        let input = input.trim();
+        if input.ends_with('%') && input.len() != 1 {
```

Comment:
You can do a
```rs
if let Some(number) = input.strip_suffix('%') { ... }
```
to avoid having to index the string in the `if` expression.

Reply:
ok